webview-ui/src/AzureServiceOperator/helpers/state.ts

```diff
@@ -41,8 +41,21 @@ function completedStep(succeeded: boolean, result: InstallStepResult | null): In
     return {status: succeeded ? InstallStepStatus.Succeeded : InstallStepStatus.Failed, result};
 }
 
-export function createState(): ASOState {
-    return {
+export type EventDef = {
```

Comment:
Note: This is removed from `userCommands.ts` and renamed from `UserMsgDef`. The original naming was intended to convey the fact that these events are typically user-initiated, in contrast to messages from VS Code. But really they are any webview events, and can be triggered programmatically, e.g. during initialization. So the name `EventDef` is more appropriately generic, and ties in more obviously with `eventHandler` on the `WebviewStateUpdater` type.